src/kinetics/InterfaceKinetics.cpp

```diff
@@ -501,64 +509,95 @@ bool InterfaceKinetics::addReaction(shared_ptr<Reaction> r_base)
-        if (re->exchange_current_density_formulation) {
-            m_has_exchange_current_density_formulation = true;
-            m_ctrxn_ecdf.push_back(1);
-        } else {
-            m_ctrxn_ecdf.push_back(0);
+        for (const auto& sp : r.reactants) {
+            size_t k = kineticsSpeciesIndex(sp.first);
+            size_t p = speciesPhaseIndex(k);
+            m_rxnPhaseIsReactant[i][p] = true;
+        }
+        for (const auto& sp : r.products) {
+            size_t k = kineticsSpeciesIndex(sp.first);
+            size_t p = speciesPhaseIndex(k);
+            m_rxnPhaseIsProduct[i][p] = true;
         }
-    }
 
-    if (r.reversible) {
-        m_revindex.push_back(i);
     } else {
-        m_irrev.push_back(i);
-    }
+        InterfaceReaction& r = dynamic_cast<InterfaceReaction&>(*r_base);
+        SurfaceArrhenius rate = buildSurfaceArrhenius(i, r, false);
+        m_rates.install(i, rate);
+        
+        // Turn on the global flag indicating surface coverage dependence
+        if (!r.coverage_deps.empty()) {
+            m_has_coverage_dependence = true;
+        }        
+        ElectrochemicalReaction* re = dynamic_cast<ElectrochemicalReaction*>(&r);
+        if (re) {
+            m_has_electrochem_rxns = true;
+            m_beta.push_back(re->beta);
+            m_ctrxn.push_back(i);
+            if (re->exchange_current_density_formulation) {
+                m_has_exchange_current_density_formulation = true;
+                m_ctrxn_ecdf.push_back(1);
+            } else {
+                m_ctrxn_ecdf.push_back(0);
+            }
+        }
+        if (r.reversible) {
+            m_revindex.push_back(i);
+        } else {
+            m_irrev.push_back(i);
+        }
 
-    m_rxnPhaseIsReactant.emplace_back(nPhases(), false);
-    m_rxnPhaseIsProduct.emplace_back(nPhases(), false);
+        m_rxnPhaseIsReactant.emplace_back(nPhases(), false);
+        m_rxnPhaseIsProduct.emplace_back(nPhases(), false);
 
-    for (const auto& sp : r.reactants) {
-        size_t k = kineticsSpeciesIndex(sp.first);
-        size_t p = speciesPhaseIndex(k);
-        m_rxnPhaseIsReactant[i][p] = true;
-    }
-    for (const auto& sp : r.products) {
-        size_t k = kineticsSpeciesIndex(sp.first);
-        size_t p = speciesPhaseIndex(k);
-        m_rxnPhaseIsProduct[i][p] = true;
+        for (const auto& sp : r.reactants) {
+            size_t k = kineticsSpeciesIndex(sp.first);
+            size_t p = speciesPhaseIndex(k);
+            m_rxnPhaseIsReactant[i][p] = true;
+        }
+        for (const auto& sp : r.products) {
+            size_t k = kineticsSpeciesIndex(sp.first);
+            size_t p = speciesPhaseIndex(k);
+            m_rxnPhaseIsProduct[i][p] = true;
+        }
     }
-
     deltaElectricEnergy_.push_back(0.0);
     m_deltaG0.push_back(0.0);
     m_deltaG.push_back(0.0);
-    m_ProdStanConcReac.push_back(0.0);
-
+    m_ProdStanConcReac.push_back(0.0); 
     return true;
```

Comment:
The non-meaningful whitespace changes here distract from the actual changes in other parts of this file.